device/iot-device-samples/device-reconnection-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/SampleParameters.java

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package samples.com.microsoft.azure.sdk.iot;
+
+import com.azure.core.http.policy.HttpLogDetailLevel;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+
+import java.net.URISyntaxException;
+import java.io.*;
+
+
+public class SampleParameters {
+
+    private static final String _DEVICE_CONNECTION_STRING = System.getenv("IOTHUB_DEVICE_CONNECTION_STRING");
+    private static final String _TRANSPORT = "mqtt";
+    private static final String _NUM_REQUESTS = "3";
+    private static final String _SLEEP_DURATION_IN_SECONDS = "10";
+    private static final String _TIMEOUT_IN_MINUTES = "1";
+
+    private static final String _FOOTER = "\nFor more info, please refer to https://github.com/Azure/azure-iot-sdks";
+    private static final String _APPEXE = "java -jar ";
+
+    private static CommandLine cmd = null;
+    private static String[] connections = new String[2];
+
+    /**
+     * Setup parameters from command line arguments
+     * @param string array from main()
+     */
+    public SampleParameters(String[] args) {
+
+        //create options for cli
+        Options options = new Options()
+            .addOption(
+                Option.builder("h")
+                    .longOpt("help")
+                    .hasArg(false)
+                    .desc("Prints this message")
+                    .build()
+            )
+            .addOption(
+                Option.builder("p")
+                    .longOpt("primaryConnectionString")
+                    .hasArg()
+                    .desc("Primary device connection string; required argument unless setup with environment variable \"IOTHUB_DEVICE_CONNECTION_STRING\"")
+                    .build()
+            )
+            .addOption(
```

Comment:
Why are we asking for the user to provide the secondary connection string? Even if it is optional, I don't see it being used anywhere in the sample.

Reply:
Was considering having the same 'parameter names' for all sample to mean the same and try to show 'secondary connection' given there is 'primary'. But agree on getting the 'help' confused event if it is optional. Will remove.